python/ray/data/dataset.py

```diff
@@ -3786,6 +3763,15 @@ def default_batch_format(self) -> Type:
                 return np.ndarray
             return pd.DataFrame
 
+    def _is_tensor_dataset(self) -> bool:
+        """Return ``True`` if this dataset is a tensor dataset."""
+        from ray.air.constants import TENSOR_COLUMN_NAME
+
+        schema = self.schema(fetch_if_missing=True)
```

Comment:
On second thought, let's keep this cheap since we don't want to make the iterator blocking.

Reply:
```suggestion
        schema = self.schema()
```